frontend/src/pages/PostList.tsx

```diff
@@ -1,29 +1,46 @@
+import { Pagination } from '@/components/Pagination';
 import { Button } from '@/components/ui/button';
 import { usePosts } from '@/hooks/usePosts';
 import { formatDate } from '@/utils';
-import { Link } from 'react-router';
+import { MAX_POSTS_PER_PAGE } from '@/utils/consts';
+import { Link, useSearchParams } from 'react-router';
 
 export default function PostList() {
-  const { postList, isLoading, error } = usePosts();
+  const [searchParams] = useSearchParams();
+
+  const pageParam = searchParams.get('page');
+  const currentPage = pageParam ? Math.max(1, parseInt(pageParam, 10)) : 1;
```

Comment:
The `Math.max(1, parseInt(pageParam, 10))` could be encapsulated within the `usePosts`, leaving the component scope much lighter. Also increasing reusability.